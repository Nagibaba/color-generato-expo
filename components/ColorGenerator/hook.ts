import { useState } from 'react';

const useContainer = () => {
  const [color, setColor] = useState('white');
  const generateColor = () => {
    const colorString = Math.random().toString(16);
    const color = '#' + colorString.substring(colorString.length - 6);
    setColor(color);
  };

  return { color, generateColor };
};

export default useContainer;
