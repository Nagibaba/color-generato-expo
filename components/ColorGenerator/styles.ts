import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  clickable: {
    flexGrow: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    fontSize: 32,
  },
});

export default styles;
