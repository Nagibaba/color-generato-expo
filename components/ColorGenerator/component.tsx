import { Stack } from 'expo-router';
import { useState } from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import useContainer from './hook';
import styles from './styles';

const ColorGenerator: React.FC = () => {
  const { color, generateColor } = useContainer();

  return (
    <TouchableOpacity
      testID="ScreenBtn"
      onPress={generateColor}
      style={[styles.clickable, { backgroundColor: color }]}
      activeOpacity={1}
    >
      <Text style={styles.text}>Hello there</Text>
    </TouchableOpacity>
  );
};

export default ColorGenerator;
