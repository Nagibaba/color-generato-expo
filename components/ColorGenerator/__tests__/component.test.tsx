import * as React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';

import ColorGenerator from '../component';

const mockHookResult = { color: jest.fn(), generateColor: jest.fn() };

jest.mock('../hook', () => jest.fn(() => mockHookResult));

describe('ColorGenerator', () => {
  it(`matches snapshot`, () => {
    const tree = render(<ColorGenerator />);

    expect(tree).toMatchSnapshot();
  });
  it(`clicking on screen runs generateColor fn`, () => {
    const tree = render(<ColorGenerator />);

    const btn = screen.getByTestId('ScreenBtn');

    fireEvent.press(btn);

    expect(mockHookResult.generateColor).toHaveBeenCalled();
  });
});
