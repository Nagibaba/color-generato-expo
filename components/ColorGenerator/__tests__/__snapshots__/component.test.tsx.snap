// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorGenerator matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": [MockFunction],
      "flexGrow": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="ScreenBtn"
>
  <Text
    style={
      {
        "fontSize": 32,
      }
    }
  >
    Hello there
  </Text>
</View>
`;
